src/coreclr/tools/aot/crossgen2/CommandLineOptions.cs

```diff
@@ -230,6 +236,162 @@ public CommandLineOptions(string[] args)
 
                 HelpText = argSyntax.GetHelpText();
             }
+
+            if (MakeReproPath != null)
+            {
+                // Create a repro package in the specified path
+                // This package will have the set of input files needed for compilation
+                // + the original command line arguments
+                // + a rsp file that should work to directly run out of the zip file
+
+                string makeReproPath = MakeReproPath;
+                Directory.CreateDirectory(makeReproPath);
+
+                List<string> crossgenDetails = new List<string>();
+                crossgenDetails.Add("CrossGen2 version");
+                try
+                {
+                    crossgenDetails.Add(Environment.GetCommandLineArgs()[0]);
+                } catch  {}
+                try
+                {
+                    crossgenDetails.Add(System.Diagnostics.FileVersionInfo.GetVersionInfo(Environment.GetCommandLineArgs()[0]).ToString());
+                } catch  {}
+
+                crossgenDetails.Add("------------------------");
+                crossgenDetails.Add("Actual Command Line Args");
+                crossgenDetails.Add("------------------------");
+                crossgenDetails.AddRange(args);
+                foreach (string arg in args)
+                {
+                    if (arg.StartsWith('@'))
+                    {
+                        string rspFileName = arg.Substring(1);
+                        crossgenDetails.Add("------------------------");
+                        crossgenDetails.Add(rspFileName);
+                        crossgenDetails.Add("------------------------");
+                        try
+                        {
+                            crossgenDetails.AddRange(File.ReadAllLines(rspFileName));
+                        } catch  {}
+                    }
+                }
+
+                HashCode hashCodeOfArgs = new HashCode();
+                foreach (string s in crossgenDetails)
+                    hashCodeOfArgs.Add(s);
+
+                string zipFileName = ((uint)hashCodeOfArgs.ToHashCode()).ToString();
+
+                if (OutputFilePath != null)
+                    zipFileName = zipFileName + "_" + Path.GetFileName(OutputFilePath);
```

Comment:
Nit: If there are multiple repro files in the same directory, it might be easier to search through them if we put `OutputFilePath` at the beginning.